packages/studio/src/store/slices/pages/createPageSlice.ts

```diff
@@ -160,6 +160,79 @@ export const createPageSlice: SliceCreator<PageSlice> = (set, get) => {
         (component) => component.uuid === activeComponentUUID
       );
     },
+    addSelectedComponentUUID: (selectedUUID: string) => {
+      set((store) => {
+        store.selectedComponentUUIDs.add(selectedUUID);
+      });
+    },
+    addSelectedComponentRect: (
+      selectedUUID: string,
+      rect: DOMRectProperties
+    ) => {
+      set((store) => {
+        store.selectedComponentRectsMap[selectedUUID] = rect;
+      });
+    },
+    clearSelectedComponents: () => {
+      set((store) => {
+        store.selectedComponentUUIDs = new Set();
+        store.selectedComponentRectsMap = {};
+      });
+    },
+    /**
+     * Adds all of the components and their children ordered between the selected component and
+     * the active component (inclusive) to selectedComponentUUIDs.
+     */
+    addShiftSelectedComponentUUIDs: (selectedComponent: ComponentState) => {
+      const selectedUUID = selectedComponent.uuid;
+      const {
+        activeComponentUUID,
+        getActivePageState,
+        addSelectedComponentUUID,
+      } = get();
+      const activePageState = getActivePageState();
+      const componentTree = activePageState?.componentTree;
+      if (selectedUUID === activeComponentUUID) return;
```

Comment:
In this case, shouldn't the active component be set as the only selected component, instead of having no selected components?

Reply:
Done